File: Python/Count_the_Characters_V3.py
def count_characters(input_str, character_input):
    count = 0
    for ch in input_str:
        if ch.lower() == character_input.lower():
            count += 1
    return count


print("Count the Characters v3")
word = input("Enter a string:")

if word == "":
    print("No input, please input a string.")
else:
    letter = input("Enter a character to count in [" + word + "]:")
    if letter == "":
        print("No input, please input a character to count.")
    else:
        final_letter = letter[0]
        result = count_characters(word, final_letter)
        print(letter + " occurs " + str(result) + " times in " + word + ".")

File: C#/Count_the_Characters_V3.cs
using System;

class Characters
{
    public int CountCharacters(string input, char characterInput)
    {
        int count = 0;
        int length = input.Length;

        for (int counter = 0; counter < length; counter++)
        {
            char container = input[counter];
            char smallContainer = char.ToLower(container);
            char smallCharacter = char.ToLower(characterInput);
            if (smallContainer == smallCharacter)
            {
                count++;
            }
        }

        return count;
    }

    static void Main(string[] args)
    {
        Characters characters = new Characters();
        Console.WriteLine("Count the Characters v3");
        Console.Write("Enter a string:");
        string word = Console.ReadLine();

        if (word == "")
        {
            Console.WriteLine("No input, please input a string.");
        }
        else
        {
            Console.Write("Enter a character to count in [" + word + "]:");
            string letter = Console.ReadLine();

            if (string.IsNullOrEmpty(letter))
            {
                Console.WriteLine("No input, please input a character to count.");
            }
            else
            {
                char finalLetter = letter[0];
                int result = characters.CountCharacters(word, finalLetter);
                Console.WriteLine(letter + " occurs " + result + " times in " + word + ".");
            }
        }
    }
}
